Assets/MixedRealityToolkit/Inspectors/Profiles/BaseMixedRealityProfileInspector.cs

```diff
@@ -69,41 +72,21 @@ private static bool RenderProfileInternal(SerializedProperty property, GUIConten
             // If it isn't, issue a warning.
             if (serviceType != null && oldObject != null)
             {
-                bool profileTypeIsValid = false;
-
-                foreach (MixedRealityServiceProfileAttribute serviceProfileAttribute in oldObject.GetType().GetCustomAttributes(typeof(MixedRealityServiceProfileAttribute), true))
-                {
-                    if (serviceProfileAttribute.ServiceType.IsAssignableFrom(serviceType))
-                    {
-                        profileTypeIsValid = true;
-                        break;
-                    }
-                }
-
-                if (!profileTypeIsValid)
+                if (!IsProfileForService(oldObject.GetType(), serviceType))
                 {
                     EditorGUILayout.HelpBox("This profile is not supported for " + serviceType.Name + ". Using an unsupported service may result in unexpected behavior.", MessageType.Warning);
                 }
             }
 
             EditorGUILayout.BeginHorizontal();
-
-            if (guiContent == null)
-            {
-                EditorGUILayout.PropertyField(property);
-            }
-            else
-            {
-                EditorGUILayout.PropertyField(property, guiContent);
-            }
-
+            RenderProfileField(property, guiContent, GetProfileTypesForService(serviceType));
             if (property.objectReferenceValue == null)
             {
-                if (showAddButton)
+                var profileTypeName = property.type.Replace("PPtr<$", string.Empty).Replace(">", string.Empty);
```

Comment:
This can be likely be simplified to property.type.Name.

just a note, _not_ needed for this PR as this is pre-existing code.

Reply:
Will take a look at this for a future change. The type is a bit... funky.